sklearn/datasets/tests/test_samples_generator.py

```diff
@@ -385,3 +386,23 @@ def test_make_moons():
         dist_sqr = ((x - center) ** 2).sum()
         assert_almost_equal(dist_sqr, 1.0,
                             err_msg="Point is not on expected unit circle")
+
+
+def test_make_circles():
+    f = 0.3
+    X, y = make_circles(7, shuffle=False, noise=None, factor=f)
+    assert_equal(X.shape, (7, 2), "X shape mismatch")
+    assert_equal(y.shape, (7,), "y shape mismatch")
+    center = [0.0, 0.0]
+    for x, label in zip(X, y):
+        dist_sqr = ((x - center) ** 2).sum()
+        dist_exp = 1.0 if label == 0 else f**2
+        assert_almost_equal(dist_sqr, dist_exp,
+                            err_msg="Point is not on expected circle")
+
+    X, y = make_circles(10, shuffle=False, noise=None)
+    assert_equal(X.shape, (10, 2), "X shape mismatch")
+    assert_equal(y.shape, (10,), "y shape mismatch")
+
+    assert_equal(X[y == 0].shape, (5, 2),
```

Comment:
If you want you could add a test that it's 3 and 4 for the test above? maybe that it's at most one different from 7 // 2? but fine either way.